cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -192,17 +193,34 @@ func Run(s *options.CMServer) error {
 		return err
 	}
 
-	// TODO: enable other lock types
-	rl := resourcelock.EndpointsLock{
-		EndpointsMeta: metav1.ObjectMeta{
-			Namespace: "kube-system",
-			Name:      "kube-controller-manager",
-		},
-		Client: leaderElectionClient,
-		LockConfig: resourcelock.ResourceLockConfig{
-			Identity:      id,
-			EventRecorder: recorder,
-		},
+	var rl resourcelock.Interface
+	switch s.LockType {
```

Comment:
I think you could probably plumb it through https://github.com/kubernetes/kubernetes/blob/master/pkg/client/leaderelection/leaderelection.go#L291